File: Domain/Entites/Product.cs

using System.ComponentModel.DataAnnotations;

namespace Domain.Entites
{
    public class Product
    {
        public int Id { get; set; }
        public string Sku { get; set; }  
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } 
        public string? Description { get; set; }
        public decimal Price { get; set; } 
        public DateTime CreatedAtUtc { get; set; }
        public DateTime? UpdatedAtUtc { get; set; }
        public bool IsDeleted { get; set; }  

    }
}

File: Client/MyTask/src/app/shared/models/product.ts
export interface CreateProduct{ 
  name:string
  sku: string
  description: string,
  price: number,  
}


export interface UpdateProduct{
  name: string
  sku: string
  description: string,
  price: number,  
}

export interface Product{
    id: number,
    sku: string,
    name: string,
    description: string,
    price: number,
    createdAtUtc: Date,
    updatedAtUtc:Date
}

export interface ProductQuery {
  pageIndex?: number;
  pageSize?: number;
  search?: string;
  includeDeleted:boolean
}

export interface PagedResult{
    productsDtos: Product[]
    pageIndex: number,
    pageSize: number,
    totalCount: number
    search:string
}

